connect/clients/ipfs.py

```diff
@@ -0,0 +1,146 @@
+"""
+ipfs.py
+
+IPFS client services that support REST calls for persistence of records with an
+IPFS cluster
+"""
+import logging
+import httpx
+import json
+from tempfile import NamedTemporaryFile
+from connect.config import get_settings
+
+
+logger = logging.getLogger(__name__)
+_ipfs_client = None
+
+
+# ******************************************
+# IPFS client services for persistence
+# ******************************************
+class IPFSClient:
+    def __init__(self, uri, replication_factor):
+        self._uri = uri
+        self._replication_factor = replication_factor
+        self._add_uri = self._uri + "/add"
+        self._get_peers_uri = self._uri + "/peers"
+        self._unpin_cid_uri = self._uri + "/pins/"
+
+    async def persist_json_to_ipfs(self, payload_dict):
+        """
+        Allows persistence of JSON payloads (as a dict) into an IPFS cluster
+        The IPFS cluster instance being connected to is configurable.
+
+        Content is pinned by default during the add operation.
+
+        :param payload_dict
+        :return content_cid: A unique ID addressing the content added to the
+                             IPFS cluster. Returns None if a non 200 response
+                             code is returned from the IPFS Cluster.
+        """
+        if not payload_dict:
+            error_msg = "dict payload not provided for persistence into IPFS"
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        async with httpx.AsyncClient() as client:
+            try:
+                payload_temp_file = NamedTemporaryFile()
```

Comment:
Let's use a context manager here so that we don't have explicitly close the file. This also ensures that in the event of an exception we will close the file as well.